File: JavaScript/Math-Array-String/47.js
// 1317. Convert Integer to the Sum of Two No-Zero Integers


/**
 * @param {number} n
 * @return {number[]}
 */
var getNoZeroIntegers = function(n) {
    // Helper function to check if a number is a No-Zero integer
    const isNoZero = (num) => {
        return !num.toString().includes('0');
    };

    // Find two No-Zero integers that sum to n
    for (let a = 1; a < n; a++) {
        let b = n - a;
        if (isNoZero(a) && isNoZero(b)) {
            return [a, b];
        }
    }
};

console.log(getNoZeroIntegers(2));
console.log(getNoZeroIntegers(11));

File: Python/Math-Array-String/47.py
# 1317. Convert Integer to the Sum of Two No-Zero Integers


from typing import List

class Solution:
    def getNoZeroIntegers(self, n: int) -> List[int]:
        def is_no_zero(num):
            # Check if a number is a No-Zero integer
            return '0' not in str(num)
        
        for a in range(1, n):
            b = n - a
            if is_no_zero(a) and is_no_zero(b):
                return [a, b]

sol = Solution()
output = sol.getNoZeroIntegers(11)
print(output)